ui_framework/components/tabs/tabs.js

```diff
@@ -0,0 +1,61 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+import classNames from 'classnames';
+import { KuiTab } from './tab';
+
+export const KuiTabs = ({
+  children,
+  selectedTabIndex,
+  onSelectedTabChanged,
+  className,
+  ...rest
+}) => {
+  const classes = classNames('kuiTabs', className);
+  const tabs = children.map((child,index)=>(
+    <KuiTab
+      key={index}
+      onClick={() => onSelectedTabChanged(index)}
+      isSelected={index === selectedTabIndex}
+    >
+      {child}
+    </KuiTab>
+  ));
+
+  return (
+    <div
+      className={classes}
+      {...rest}
+    >
+      {tabs}
+    </div>
+  );
+};
+
+const selectedTabIndexCheck = (props, propName, componentName) => {
+  const childrenLength = props.children ? props.children.length : 0;
+  const selectedTabIndex = props.selectedTabIndex;
```

Comment:
Aha. You think about re-use. I would do it in a later run when there are enough patterns to abstract.
Until that better so simple to read as possible and that means staying so close to React itself as possible.